src/htr2hpc/train_apps.py

```diff
@@ -0,0 +1,45 @@
+import pathlib
+from zipfile import ZipFile
+
+import parsl
+from parsl.app.app import python_app, bash_app
+from parsl.data_provider.files import File
+
+from htr2hpc.api_client import eScriptoriumAPIClient
+
+
+@python_app
+def prep_training_data(es_base_url, es_api_token, document_id, transcription_id):
+    from htr2hpc.api_client import eScriptoriumAPIClient
+
+    api = eScriptoriumAPIClient(es_base_url, api_token=es_api_token)
+    # get current user info, since id is needed to determine export filename
+    user = api.current_user()
+    # get document details so we don't have to specify both document id and name
+    document = api.document(document_id)
+    export_zipfile = api.download_document_export(
+        user.pk, document_id, document.name, transcription_id
+    )
+    # run in current working directory for now
+    data_dir = pathlib.Path("training_data")
+    data_dir.mkdir(exist_ok=True)
+
+    # extract everything in the zipfiile to the training data dir
+    with ZipFile(export_zipfile) as zip_exp:
+        zip_exp.extractall(path=data_dir)
+
+    # return training directory as a parsl file
+    return File(data_dir)
+
+
+@bash_app
+def segtrain(inputs=[], outputs=[]):
```

Comment:
We could put both transcription and segmentation training in one function, but we probably want a separate `train` function for transcription training. I'm laying out the usual eScriptorium train/segtrain commands as they would be formatted in command-line kraken below.

- A command for transcription training from scratch:

  - `ketos train --epochs 50 -o model --workers HOW_MANY_CPUS -d cuda:0 -f alto {input_data_dir}/*.xml`
  - (For workers, use the number of CPUs requested in the slurm job.)
- A command for transcription refining on another model:
  - `ketos train --epochs 50 --resize union -i {path_to_starting_model} -o model --workers HOW_MANY_CPUS -d cuda:0 -f alto {input_data_dir}/*.xml`
- A command for segmentation training "from scratch":
  - `ketos segtrain --epochs 50 --resize both -i {path_to_blla_mlmodel} -o model --workers HOW_MANY_CPUS -d cuda:0 -f xml {input_data_dir}/*.xml`
- A command for segmentation refining on another model:
  - `ketos segtrain --epochs 50 --resize both -i {path_to_starting_model} -o model --workers HOW_MANY_CPUS -d cuda:0 -f xml {input_data_dir}/*.xml`